sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -374,26 +422,12 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(OSModuleHandle M,
 
   std::string CompileOpts;
   std::string LinkOpts;
-  // Build options are overridden if environment variables are present.
-  // Environment variables are not changed during program lifecycle so it
-  // is reasonable to use static here to read them only once.
-  static const char *CompileOptsEnv =
-      SYCLConfig<SYCL_PROGRAM_COMPILE_OPTIONS>::get();
-  if (CompileOptsEnv) {
-    CompileOpts = CompileOptsEnv;
-  } else { // Use build options only when the environment variable is missed
-    if (Prg) {
-      std::string BuildOptions = Prg->get_build_options();
-      if (!BuildOptions.empty()) {
-        CompileOpts += " ";
-        CompileOpts += BuildOptions;
-      }
-    }
-  }
-  static const char *LinkOptsEnv = SYCLConfig<SYCL_PROGRAM_LINK_OPTIONS>::get();
-  if (LinkOptsEnv) {
-    LinkOpts = LinkOptsEnv;
+  if (Prg) {
+    CompileOpts = Prg->get_build_options();
   }
+
+  applyOptionsFromEnvironment(CompileOpts, LinkOpts);
```

Comment:
Is it possible to unify `applyOptionsFromEnvironment` and `applyOptionsFromImage`?

Reply:
I thought about it but have not find the way:
 - applyOptionsFromEnvironment reads only environment and is used before build lambda to pass result to program cache key generator.
 - applyOptionsFromImage reads image which is available only in build lambda (getting it before lambda causes performance degradation in applications with many calls of small kernels)